raster/rt_pg/rtpg_geometry.c

```diff
@@ -651,6 +654,324 @@ Datum RASTER_getPixelPolygons(PG_FUNCTION_ARGS)
 	}
 }
 
+#undef VALUES_LENGTH
+#define VALUES_LENGTH 4
+
+/**
+ * Return centroid points of all pixels
+ */
+PG_FUNCTION_INFO_V1(RASTER_getPixelCentroids);
+Datum RASTER_getPixelCentroids(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	TupleDesc tupdesc;
+	rt_pixel pix = NULL;
+	rt_pixel pix2;
+	int call_cntr;
+	int max_calls;
+	int i = 0;
+
+	if (SRF_IS_FIRSTCALL()) {
+		MemoryContext oldcontext;
+		rt_pgraster *pgraster = NULL;
+		rt_raster raster = NULL;
+		rt_band band = NULL;
+		int nband = 1;
+		int numbands;
+		bool hasband = FALSE;
+		bool exclude_nodata_value = TRUE;
+		bool nocolumnx = TRUE;
+		bool norowy = TRUE;
+		int x = 0;
+		int y = 0;
+		int bounds[4] = {0};
+		int pixcount = 0;
+		double value = 0;
+		int isnodata = 0;
+
+		LWPOINT* point = NULL;
+
+		POSTGIS_RT_DEBUG(3, "RASTER_getPixelCentroids first call");
+
+		/* create a function context for cross-call persistence */
+		funcctx = SRF_FIRSTCALL_INIT();
+
+		/* switch to memory context appropriate for multiple function calls */
+		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+
+		/* raster */
+		if (PG_ARGISNULL(0)) {
+			MemoryContextSwitchTo(oldcontext);
+			SRF_RETURN_DONE(funcctx);
+		}
+		pgraster = (rt_pgraster *) PG_DETOAST_DATUM(PG_GETARG_DATUM(0));
+
+		/* band */
+		if (!PG_ARGISNULL(1)) {
+			nband = PG_GETARG_INT32(1);
+			hasband = TRUE;
+		}
+
+		/* column */
+		if (!PG_ARGISNULL(2)) {
+			bounds[0] = PG_GETARG_INT32(2);
+			bounds[1] = bounds[0];
+			nocolumnx = FALSE;
+		}
+
+		/* row */
+		if (!PG_ARGISNULL(3)) {
+			bounds[2] = PG_GETARG_INT32(3);
+			bounds[3] = bounds[2];
+			norowy = FALSE;
+		}
+
+		/* exclude NODATA */
+		if (!PG_ARGISNULL(4))
+			exclude_nodata_value = PG_GETARG_BOOL(4);
+
+		/* deserialize raster */
+		raster = rt_raster_deserialize(pgraster, FALSE);
+		if (!raster) {
+			PG_FREE_IF_COPY(pgraster, 0);
+			ereport(ERROR, (
+				errcode(ERRCODE_OUT_OF_MEMORY),
+				errmsg("Could not deserialize raster")
+			));
+			MemoryContextSwitchTo(oldcontext);
+			SRF_RETURN_DONE(funcctx);
+		}
+
+		/* raster empty, return NULL */
+		if (rt_raster_is_empty(raster)) {
+			elog(NOTICE, "Raster is empty. Returning NULL");
```

Comment:
technically it's not a null but an empty set?

Reply:
This was copied from `RASTER_getPixelPolygons`, `rtpg_geometry.c:440` and a couple more, should I change these to "Returning empy set" too?

Reply:
yes, either fix or just erase this debug print, if it was wrong for years and nobody noticed there's no value in it.

Reply:
Done.